src/raftstore/store/debug.rs

```diff
@@ -151,6 +166,225 @@ impl Debugger {
             Err(e) => Err(box_err!(e)),
         }
     }
+
+    pub fn scan_mvcc<D, E>(&self, req: ScanMvccRequest, deal: &mut D) -> Result<()>
+    where
+        Error: ::std::convert::From<E>,
+        D: FnMut(Vec<u8>, MvccInfo) -> ::std::result::Result<(), E>,
+    {
+        let from_key = keys::data_key(req.get_from_key());
+        let limit = req.get_limit();
+        let to_key = Some(req.get_to_key())
+            .into_iter()
+            .filter(|k| !k.is_empty())
+            .map(|k| keys::data_key(k))
+            .next();
+        if to_key.is_none() && limit == 0 {
+            return Err(Error::InvalidArgument("no limit and to_key".to_owned()));
+        }
+
+        let lock_rx = self.mvcc_kvs(CF_LOCK, from_key.clone(), to_key.clone());
+        let default_rx = self.mvcc_kvs_grouped(CF_DEFAULT, from_key.clone(), to_key.clone());
+        let write_rx = self.mvcc_kvs_grouped(CF_WRITE, from_key.clone(), to_key.clone());
+
+        let mut mvcc_infos: BTreeMap<Vec<u8>, (MvccInfo, bool, bool, bool)> = BTreeMap::new();
+        let (mut want_lock, mut want_default, mut want_write) = (true, true, true);
+        let (mut m_l, mut m_d, mut m_w) = (None, None, None);
+        let mut count = 0;
+        loop {
+            if !want_lock && !want_default && !want_write {
+                break;
+            }
+            if want_lock {
+                if let Ok(kv) = lock_rx.recv() {
```

Comment:
Maybe we can use `select!()` [1].

[1] https://doc.rust-lang.org/std/macro.select.html